apps/src/templates/projects/ReportAbusePopUp.jsx

```diff
@@ -0,0 +1,279 @@
+          : checkbox
+      );
+
+      const showRecaptcha = updatedCheckboxes.some(
+        checkbox => checkbox.checked
+      );
+
+      return {
+        checkboxes: updatedCheckboxes,
+        showRecaptcha,
+      };
+    });
+  };
+
+  createCaptchaScript() {
+    const script = document.createElement('script');
+    script.src = 'https://www.google.com/recaptcha/api.js';
+    script.id = 'captcha';
+    script.async = true;
+    script.defer = true;
+    window.onCaptchaSubmit = token => this.onCaptchaVerification(token);
+    window.onCaptchaExpired = () => this.onCaptchaExpiration();
+    script.onload = () => this.setState({loadedCaptcha: true});
+    document.body.appendChild(script);
+  }
+
+  render() {
+    const {
+      showRecaptcha,
+      checkboxes,
+      submitButtonEnabled,
+      showReportConfirmation,
+    } = this.state;
+
+    const captchaSiteKey = this.props.recaptchaSiteKey;
+
+    return (
+      <AccessibleDialog className={style.popUp} onClose={this.cancel}>
+        {showReportConfirmation ? (
+          <div className={style.submitConfirmation}>
+            <h3>{i18n.thankyou()}!</h3>
+            <p>{i18n.thankYouForReport()}</p>
+            <Button
+              onClick={this.cancel}
+              text={'Continue'}
+              color={Button.ButtonColor.brandSecondaryDefault}
+            />
+          </div>
+        ) : (
+          <div>
+            <div className={style.title}>
+              <h3 style={{margin: 0}}>{i18n.reportAbuse()}</h3>
+              <button
+                type="reset"
+                onClick={this.cancel}
+                className={style.xButton}
+              >
+                <FontAwesome icon="x" className={style.xIcon} />
+              </button>
+            </div>
+            <hr className={style.lines} />
+            <p className={style.body}>{i18n.whyReport()}</p>
+            <div>
+              {checkboxes.map(checkbox => (
+                <CheckBox
+                  key={checkbox.key}
+                  label={checkbox.label}
+                  checked={checkbox.checked}
+                  onChange={() => this.handleCheckboxChange(checkbox.key)}
+                  size="s"
+                />
+              ))}
+            </div>
+            {showRecaptcha ? (
+              <div
+                style={{padding: 5}}
+                className="g-recaptcha"
+                data-sitekey={captchaSiteKey}
+                data-callback="onCaptchaSubmit"
+                data-expired-callback="onCaptchaExpired"
+              />
+            ) : null}
+            <hr className={style.lines} />
+            <div className={style.buttonHolder}>
+              <Button
+                onClick={this.cancel}
+                text={i18n.cancel()}
+                color={Button.ButtonColor.neutralDark}
+              />
+              <Button
+                onClick={this.handleSubmit}
+                disabled={!submitButtonEnabled}
```

Comment:
I can submit an abuse report without a reason currently, if you check a checkbox -> submit captcha -> uncheck checkbox:

![image](https://github.com/code-dot-org/code-dot-org/assets/25372625/02275a9c-ef5f-45cf-8d96-343d689fce5e)

I might suggest that `submitButtonEnabled` doesn't need to be its own bit of state, and you could just derive whether to disable the button based on the # of checkboxes checked, and the current state of captcha completion (ie, `disabled={noCheckboxesChecked || !captchaCompleted}`). Looks like you also disable the button currently if there's an error submitting, but not sure if that's totally necessary.

Reply:
Nice suggestion!